tests/test_scheduler.py

```diff
@@ -2333,3 +2318,79 @@ def test_scheduler_disable_wildcard(here, config):
         dirname.rmdir()
     except FileNotFoundError:
         pass
+
+
+def test_transformation_config(config):
+    """
+    Test the correct instantiation of :any:`Transformation` objecst from config
+    """
+    my_config = config.copy()
+    my_config['transformations'] = {
+        'DependencyTransformation': {
+            'module': 'loki.transform',
+            'options':
+            {
+                'suffix': '_rick',
+                'module_suffix': '_roll',
+                'replace_ignore_items': False,
+            }
+        }
+    }
+    cfg = SchedulerConfig.from_dict(my_config)
+    assert cfg.transformations['DependencyTransformation']
+
+    transformation = cfg.transformations['DependencyTransformation']
+    assert isinstance(transformation, DependencyTransformation)
+    assert transformation.suffix == '_rick'
+    assert transformation.module_suffix == '_roll'
+    assert not transformation.replace_ignore_items
+
+    # Test for errors when failing to instantiate a transformation
+    bad_config = config.copy()
+    bad_config['transformations'] = {
+        'DependencyTrafo': {  # <= typo
+            'module': 'loki.transform',
+            'options': {}
+        }
+    }
+    with pytest.raises(RuntimeError):
+        cfg = SchedulerConfig.from_dict(bad_config)
+
+    worse_config = config.copy()
+    worse_config['transformations'] = {
+        'DependencyTransform': {
+            'module': 'loki.transformats',  # <= typo
+            'options': {}
+        }
+    }
+    with pytest.raises(ModuleNotFoundError):
+        cfg = SchedulerConfig.from_dict(worse_config)
```

Comment:
Just to point this out: For full coverage we would need a `worst_config` with invalid options provided, that's the only bit currently not covered in the tests.